node/src/test_utils/mod.rs

```diff
@@ -713,13 +715,15 @@ pub mod unshared_test_utils {
     }
 
     pub struct NotifyLaterHandleMock<T> {
-        notify_later_params: Arc<Mutex<Vec<(T, Duration)>>>, //I care only about the params; realize that it's hard to test self addressed messages if you cannot mock yourself
+        notify_later_params: Arc<Mutex<Vec<(T, Duration)>>>,
+        do_you_want_to_proceed_after: bool,
```

Comment:
When you wrote this, I bet it was really clear to you what "proceed" meant; but when another developer encounters this in a year, it won't be obvious to him.

Consider a name change to something involving sending the message on or not sending it on.